src/dowel/csv_output.py

```diff
@@ -56,6 +54,36 @@ def record(self, data, prefix=''):
         else:
             raise ValueError('Unacceptable type.')
 
+    def _augment(self, data):
+        """Augment tabular data with new column(s)"""
+        data.reset()
+        self._log_file.close()
+
+        temp_file = self._file_name.split(".")
+        temp_file = "".join(temp_file[:-1]) + "_temp." + temp_file[-1]
+        os.rename(self._file_name, temp_file)
+        old_file = open(temp_file, 'r')
+        self._log_file = open(self._file_name, 'w')
+
+        self._fieldnames = data.as_primitive_dict.keys()
+        self._writer = csv.DictWriter(
+                self._log_file,
+                fieldnames=self._fieldnames,
+                extrasaction='ignore')
```

Comment:
Why use `extrasaction='ignore'` here? It seems like `extrasaction='raise'` (the default) would be a safer choice.

Reply:
Agreed